pinot-broker/src/main/java/org/apache/pinot/broker/routing/BrokerRoutingManager.java

```diff
@@ -458,13 +461,32 @@ public boolean routingExists(String tableNameWithType) {
    * <p>NOTE: The broker request should already have the table suffix (_OFFLINE or _REALTIME) appended.
    */
   @Nullable
+  @Override
   public RoutingTable getRoutingTable(BrokerRequest brokerRequest) {
     String tableNameWithType = brokerRequest.getQuerySource().getTableName();
     RoutingEntry routingEntry = _routingEntryMap.get(tableNameWithType);
     if (routingEntry == null) {
       return null;
     }
     InstanceSelector.SelectionResult selectionResult = routingEntry.calculateRouting(brokerRequest);
+    return constructRoutingTableFromInstanceSelectionResult(tableNameWithType, selectionResult);
+  }
+
+  @Override
+  public Map<String, ServerInstance> getEnabledServerInstanceMap() {
+    return new HashMap<>(_enabledServerInstanceMap);
```

Comment:
Directly return the `_enabledServerInstanceMap` to avoid the unnecessary overhead